forest/daemon/src/daemon.rs

```diff
@@ -169,6 +169,7 @@ pub(super) async fn start(config: Config, detached: bool) -> Db {
             {
                 error!("Failed to initiate prometheus server: {e}");
             }
+            Ok(())
```

Comment:
Why ignore the error here?

Reply:
It's bad prometheus server failed to initiate, but it seems to to be critical to Forest

Reply:
We already fail if bind to the port or find the DB path. Let's fail as eagerly as possible and exit Forest if something goes wrong in prometheus. It'll then be prometheus' responsibility to properly deal with non-fatal errors.

Reply:
Fixed accordingly